### ISSUE TITLE
THNN.lua:110: input and gradOutput have different number of elements: input[20 x 26] has 520 elements, while gradOutput[26] has 26

### ISSUE THREAD

**Issue by loveJasmine** [jiasenlu/HieCoAttenVQA#20]
rzai@rzai00:~/prj/HieCoAttenVQA$ th train.lua -input_img_train_h5 data/vqa_data_img_vgg_train.h5 -input_img_test_h5 data/vqa_data_img_vgg_test.h5 -input_ques_h5 data/vqa_data_prepro.h5 -input_json data/vqa_data_prepro.json -co_atten_type Alternating -feature_type VGG 
{
  input_img_train_h5 : "data/vqa_data_img_vgg_train.h5"
  learning_rate_decay_every : 300
  optim : "rmsprop"
  hidden_size : 512
  optim_epsilon : 1e-08
  output_size : 1000
  rnn_layers : 2
  input_img_test_h5 : "data/vqa_data_img_vgg_test.h5"
  losses_log_every : 600
  id : "0"
  input_ques_h5 : "data/vqa_data_prepro.h5"
  learning_rate_decay_start : 0
  start_from : ""
  gpuid : 6
  seed : 123
  input_json : "data/vqa_data_prepro.json"
  optim_beta : 0.995
  batch_size : 20
  iterPerEpoch : 1200
  rnn_size : 512
  max_iters : -1
  checkpoint_path : "save/train_vgg"
  save_checkpoint_every : 6000
  learning_rate : 0.0004
  co_atten_type : "Alternating"
  feature_type : "VGG"
  backend : "cudnn"
  optim_alpha : 0.99
}
DataLoader loading h5 image file: 	data/vqa_data_img_vgg_train.h5	
DataLoader loading h5 image file: 	data/vqa_data_img_vgg_test.h5	
DataLoader loading h5 question file: 	data/vqa_data_prepro.h5	
DataLoader loading json file: 	data/vqa_data_prepro.json	
assigned 215375 images to split 0	
assigned 121512 images to split 2	
Building the model...	
total number of parameters in word_level: 	8031747	
total number of parameters in phrase_level: 	2889219	
total number of parameters in ques_level: 	5517315	
constructing clones inside the ques_level	
total number of parameters in recursive_attention: 	2862056	
/home/rzai/torch/install/bin/luajit: /home/rzai/torch/install/share/lua/5.1/nn/THNN.lua:110: input and gradOutput have different number of elements: input[20 x 26] has 520 elements, while gradOutput[26] has 26 elements at /home/rzai/torch/extra/cunn/lib/THCUNN/generic/SoftMax.cu:84
stack traceback:
	[C]: in function 'v'
	/home/rzai/torch/install/share/lua/5.1/nn/THNN.lua:110: in function 'SoftMax_updateGradInput'
	./misc/maskSoftmax.lua:33: in function 'updateGradInput'
	/home/rzai/torch/install/share/lua/5.1/nngraph/gmodule.lua:420: in function 'neteval'
	/home/rzai/torch/install/share/lua/5.1/nngraph/gmodule.lua:454: in function 'updateGradInput'
	/home/rzai/torch/install/share/lua/5.1/nn/Module.lua:31: in function 'backward'
	./misc/ques_level.lua:143: in function 'updateGradInput'
	/home/rzai/torch/install/share/lua/5.1/nn/Module.lua:31: in function 'backward'
	train.lua:272: in function 'lossFun'
	train.lua:310: in main chunk
	[C]: in function 'dofile'
	...rzai/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00406670
rzai@rzai00:~/prj/HieCoAttenVQA$

**Comment by Cadene** [comment]
https://github.com/jiasenlu/HieCoAttenVQA/pull/21/files

**Comment by loveJasmine** [comment]
good job!
